### PR TITLE
Add Datasource for Tags, CPUPerformanceTypes and VSphere Locations

### PR DESCRIPTION
**Pull request by stroebitzer**
### Description

Add Datasource for Tags and CPUPerformanceTypes

### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAnxCloudTagsDataSource'
$ make testacc TESTARGS='-run=TestAccAnxCloudCPUPerformanceTypesDataSource'
$ make testacc TESTARGS='-run=TestAccAnxCloudVSphereLocationsDataSource'
```

### Release Note
Release note for [CHANGELOG](https://github.com/anexia-it/terraform-provider-anxcloud/blob/master/CHANGELOG.md):

```release-note
Add Datasource for Tags
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

**Comment by mfranczy** [comment]
Unit tests fail, please run `make go-lint`, `go mod tidy` and `make docs-lint-fix`

**Comment by mfranczy** [comment]
ah ok it fails becasue of `v0.3.11` is not released yet

File: go.mod [hunk 1]
```diff
@@ -21,3 +21,5 @@ require (
 	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
 	gopkg.in/yaml.v3 v3.0.0-20200506231410-2ff61e1afc86 // indirect
 )
+
+replace github.com/anexia-it/go-anxcloud v0.3.11 => ../go-anxcloud
```
Comment (mfranczy) [inline go.mod:25]: @stroebitzer could you remove this from go.mod?

File: anxcloud/struct_vsphere_locations.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package anxcloud
```
Comment (mfranczy) [inline anxcloud/struct_vsphere_locations.go:1]: could you move that to `struct_locations.go` file to have each location in one file? That would help later with refactoring.

File: anxcloud/struct_vsphere_locations_test.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
+package anxcloud
```
Comment (mfranczy) [inline anxcloud/struct_vsphere_locations_test.go:1]: could you move that to `struct_locations_test.go` file to have each location in one file? That would help later with refactoring.

File: go.mod (hunk 1 above)
Comment (stroebitzer) [reply go.mod]: done

File: examples/main.tf [hunk 4]
```diff
@@ -9,4 +9,4 @@ terraform {
 
 provider "anxcloud" {}
 
-data "anxcloud_core_locations" "example" {}
\ No newline at end of file
+data "anxcloud_vsphere_locations" "example" {}
```
Comment (mfranczy) [inline examples/main.tf:12]: please restore the original `examples/main.tf`

**Review by mfranczy — CHANGES_REQUESTED** [review]
Looks good, just a few minor things.

File: anxcloud/struct_vsphere_locations.go (hunk 2 above)
Comment (stroebitzer) [reply anxcloud/struct_vsphere_locations.go]: done

File: anxcloud/struct_vsphere_locations_test.go (hunk 3 above)
Comment (stroebitzer) [reply anxcloud/struct_vsphere_locations_test.go]: done

File: anxcloud/data_source_vsphere_locations_test.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccAnxCloudVSphereLocationsDataSource(t *testing.T) {
+	resourceName := "acc_locations_test"
+	resourcePath := "data.anxcloud_locations." + resourceName
```
Comment (mfranczy) [inline anxcloud/data_source_vsphere_locations_test.go:13]: @stroebitzer there is one more issue, that should be `acc_vsphere_locations`

**Pull request merged**